### ISSUE TITLE
CompatibilityError in replace_in_file

### ISSUE THREAD

**Issue by jonatanlinden** [piotrmurach/tty-file#9]
### Describe the problem

When doing `teletype new app`
I get an exception in tty-file (omitting the complete stack trace):
```
7: from ~/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/tty-0.9.1/lib/tty/commands/new.rb:224:in `add_license_to_readme'
...
1: from ~/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/tty-file-0.7.0/lib/tty/file.rb:587:in `replace_in_file'
~/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/tty-file-0.7.0/lib/tty/file.rb:587:in `gsub!': incompatible character encodings: ASCII-8BIT and UTF-8 (Encoding::CompatibilityError)
```
The issue might belong to the teletype repository, but since one possible change in tty-file fixes the problem, I deemed it more relevant to put it here.

### Steps to reproduce the problem

`teletype new app`

### Actual behaviour

Got an exception, `incompatible character encodings: ASCII-8BIT and UTF-8 (Encoding::CompatibilityError)`.

### Expected behaviour

No exception generated.

By undoing commit d185080e789bd80167cd6e8aa3272b45c858d326, the expected behaviour was observed. Maybe binary reading and writing of files is important, but it doesn't seem to handle encodings correctly in all cases.

In the function `replace_in_file` in `file.rb`, I observed that `contents.encoding` returned `#<Encoding:ASCII-8BIT>`, and `args[0].encoding` returned `#<Encoding:US-ASCII>`.

### Describe your environment

* OS version: macOS 10.13.6
* Ruby version: 2.6.2
* TTY::File version: 0.7.0

**Comment by piotrmurach** [comment]
Thanks, Jonatan ❤️ I really appreciate you reporting this issue. It's fixed in the `0.7.1` that will be released shortly.

**Comment by piotrmurach** [comment]
Released `v0.7.1`
